tests/test_x2pdf.py

```diff
@@ -9,23 +9,42 @@
 from .conftest import run_cli
 
 
-def test_x2pdf(capsys, tmp_path: Path) -> None:
+def test_x2pdf_with_format(capsys, tmp_path: Path) -> None:
     # Arrange
     output = tmp_path / "out.pdf"
     assert not output.exists()
+    
+    formats_to_test = [
+        "Letter",
+        "A4-portrait",
+        "A4-landscape",
+        "210x297",
+        "invalid-format"
+    ]
+    
+    for format_option in formats_to_test:
+        # Act
+        exit_code = run_cli(
+            [
+                "x2pdf",
+                "sample-files/003-pdflatex-image/page-0-Im1.jpg",
+                "--output",
+                str(output),
+                "--format",
+                format_option,
+            ]
+        )
 
-    # Act
-    exit_code = run_cli(
-        [
-            "x2pdf",
-            "sample-files/003-pdflatex-image/page-0-Im1.jpg",
-            "--output",
-            str(output),
-        ]
-    )
-
-    # Assert
-    captured = capsys.readouterr()
-    assert exit_code == 0, captured
-    assert captured.out == ""
-    assert output.exists()
+        # Assert
+        captured = capsys.readouterr()
+        
+        # For valid formats, we expect a successful exit code and the output file to exist
+        if format_option != "invalid-format":
+            assert exit_code == 0, captured
+            assert captured.out == ""
+            assert output.exists()
```

Comment:
It would be interesting to also validate the resulting pages dimensions.

This can be checked with `pdfly pagemeta $pdf_filepath $page_index`

Or using the underlying `pypdf` library: `PdfReader(pdf: Path).mediabox/.cropbox/.artbox/.bleedbox`

Reply:
I think you did not handle that feedback comment @mulla028 🙂